ConfigurationSystem/Agent/VOMS2CSAgent.py

```diff
@@ -42,582 +34,149 @@ def __init__( self, *args, **kwargs ):
-          else:
-            oldValue = str( diracUserDict.get( diracName, {} ).get( key, '' ) )
-            if str( userDict[key] ) != oldValue:
-              modMsg += "    %s: %s -> %s\n" % ( key, oldValue, str( userDict[key] ) )
-        if modMsg:
-          self.__adminMsgs[ 'Info' ].append( message + modMsg )
-          modified = True
-
-      if self.autoModifyUsers and modified:
-        result = self.csapi.modifyUser( diracName, userDict )
-        if result['OK'] and result['Value']:
-          self.log.info( "Modified user %s: %s" % ( diracName, str( userDict ) ) )
-          self.voChanged = True
-          resultDict['ModifiedUsers'].append( diracName )
-
-    # Check if there are potentially obsoleted users
-    oldUsers = set()
-    for user in diracUserDict:
-      dnSet = set( fromChar( diracUserDict[user]['DN'] ) )
-      if not dnSet.intersection( set( vomsUserDict.keys() ) ) and user not in nonVOUserDict:
-        for group in diracUserDict[user]['Groups']:
-          if group not in noVOMSGroups:
-            oldUsers.add( user )
-
-    # Check for obsoleted DNs
-    for user in diracUserDict:
-      dnSet = set( fromChar( diracUserDict[user]['DN'] ) )
-      for dn in dnSet:
-        if dn in obsoletedDNs and user not in oldUsers:
-          self.log.verbose( "Modified user %s: dropped DN %s" % ( user, dn ) )
-          if self.autoModifyUsers:
-            userDict = diracUserDict[user]
-            modDNSet = dnSet - set( [dn] )
-            if modDNSet:
-              userDict['DN'] = ','.join( modDNSet )
-              result = self.csapi.modifyUser( user, userDict )
-              if result['OK'] and result['Value']:
-                self.log.info( "Modified user %s: dropped DN %s" % ( user, dn ) )
-                self.__adminMsgs[ 'Info' ].append( "Modified user %s: dropped DN %s" % ( user, dn ) )
-                self.voChanged = True
-                resultDict['ModifiedUsers'].append( diracName )
-            else:
-              oldUsers.add( user )
-
-
-    if oldUsers:
-      self.voChanged = True
-      if self.autoDeleteUsers:
-        self.log.info( 'The following users will be deleted: %s' % str( oldUsers ) )
-        result = self.csapi.deleteUsers( oldUsers )
-        if result['OK']:
-          self.__adminMsgs[ 'Info' ].append( 'The following users are deleted from CS:\n  %s\n' % str( oldUsers ) )
-          resultDict['DeletedUsers'] = oldUsers
-        else:
-          self.__adminMsgs[ 'Errors' ].append( 'Error in deleting users from CS:\n  %s' % str( oldUsers ) )
-          self.log.error( 'Error while user deletion from CS', result )
-      else:
-        self.__adminMsgs[ 'Info' ].append( 'The following users to be checked for deletion:\n  %s' % str( oldUsers ) )
-        self.log.info( 'The following users to be checked for deletion: %s' % str( oldUsers ) )
-
-    return S_OK( resultDict )
+  def __syncCSWithVOMS(self, vomsSync):
+    return vomsSync.syncCSWithVOMS()
```

Comment:
Do we need this method at all?

Reply:
This is a method wrapped in @executeWithUserProxy decorator